src/lib/fhirTables.js

```diff
@@ -397,3 +413,196 @@ function obsCompare(a, b) {
+    </tr>
+  );
+}
+
+function carePlanRow(r, rmap, dcr) {
+  const status = r.status;
+  const intent = r.intent;
+
+  const activities = futil.joinJSXElements(
+    (r.activity || []).map(activity =>
+      activity.detail && activity.detail.code
+      ? futil.renderCodeableJSX(activity.detail.code, dcr)
+      : null
+    ).filter(activity => activity !== null),
+    ', '
+  );
+
+  const category = futil.joinJSXElements(
+    (r.category || []).map(c =>
+      c.text ? futil.renderCodeableJSX(c, dcr) : null
+    ).filter(c => c !== null),
+    ', '
+  );
+
+  const period = r.period ? futil.renderPeriod(r.period) : "";
+
+  return (
+    <tr key={r.id}>
+      <td>{status}</td>
+      <td>{intent}</td>
+      <td>{activities}</td>
+      <td>{category}</td>
+      <td>{period}</td>
+      {/* Render other relevant CarePlan properties as table cells */}
+    </tr>
+  );
+}
+
+// +--------------------+
+// |      Consent       |
+// +--------------------+
+
+function consentHeader() {
+  return (
+    <tr>
+      <th>Status</th>
+      <th>Scope</th>
+      <th>Category</th>
+      <th>Date/Time</th>
+      <th>Policy Rule</th>
+      <th>Provision Period</th>
+      <th>Organization</th>
+    </tr>
+  );
+}
+
+function consentRow(r, rmap, dcr) {
+  const status = r.status || "N/A";
+  const scopeDisplay = r.scope && Array.isArray(r.scope.coding) && r.scope.coding.length > 0
+    ? futil.renderCodeableJSX(r.scope, dcr)
+    : "";
+  const categoryDisplay = r.category && Array.isArray(r.category) && r.category.length > 0
+    ? futil.renderCodeableJSX(r.category[0], dcr)
+    : "";
+  const dateTime = r.dateTime
+    ? futil.renderDateTime(r.dateTime)
+    : "";
+  const policyRule = r.policyRule && Array.isArray(r.policyRule.coding) && r.policyRule.coding.length > 0
+    ? futil.renderCodeableJSX(r.policyRule, dcr)
+    : "";
+  const provisionPeriod = r.provision && r.provision.period
+    ? futil.renderPeriod(r.provision.period)
+    : "";
+  const organization = r.organization && Array.isArray(r.organization) && r.organization.length > 0
+    ? futil.renderOrganization(r.organization[0], dcr)
+    : "";
+
+  return (
+    <tr key={r.id}>
+      <td>{status}</td>
+      <td>{scopeDisplay}</td>
+      <td>{categoryDisplay}</td>
+      <td>{dateTime}</td>
+      <td>{policyRule}</td>
+      <td>{provisionPeriod}</td>
+      <td>{organization}</td>
+    </tr>
+  );
+}
+
+// +--------------------+
+// | DeviceUseStatement |
+// +--------------------+
+
+function deviceUseStatementHeader() {
+    return (
+        <tr>
+            <th>Subject</th>
+            <th>Timing</th>
+            <th>Source</th>
+            <th>Device</th>
+            <th>Body Site</th>
+            {/* Add other relevant headers if needed */}
+        </tr>
+    );
+}
+
+function deviceUseStatementRow(r, rmap, dcr) {
+    const subject = futil.renderReference(r.subject, dcr);
+    let timing = null;
+
+    if (r.timingDateTime) {
```

Comment:
there's another utility for these elements that use different types like timing[x] ... you can call futil.renderCrazyDateTime(r. "timing") and it will grovel around in there looking for things like timingDateTime, timingPeriod, etc. etc.

HOWEVER I just noticed that when I originally wrote renderCrazyDateTime, "Timing" was NYI and it will render as unsupported. But we now have a renderTiming function, so if you go to fhirUtil.js at the bottom of renderCrazyDateTime, you can just change that "unsupported" clause to call the renderTiming function and then it will work!